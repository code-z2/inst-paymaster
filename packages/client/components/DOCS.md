# Paymaster

This is a description of the components , stylesheets and
pages that makes up paymaster project.

## Components

This project makes use of tailwindcss and css modules.

Each component is styled using a local stylesheet. This makes it easy to identify where to make changes for any section or component in the page.

The navbar/footer component and other global variables are styled using a global stylesheet.

Components bear similar names to it's stylesheet modules
for easy identification.

E.g `landingpage/firstSection` styling can be seen in `styles/landing/firstsection.module.css`

Adopted Css grid method [Multi-grid-one-page-layout](https://medium.com/@nikkipantony/multi-grid-one-page-layout-css-grid-6efefd537404);

The components in this directory are:

## Active paymaster

- **activeNav**: This is the navbar of the active paymaster's page.
  `activepaymaster/activeNav`
  - styling: open `styles/navbar.css` to view the navbar styling.
    **All the navbars in the project are styled from**
    `styles/navbar.css`

```
 .active-paymaster-nav {
    @apply paynav;
    }

.active-paymaster-nav .list-ul {
@apply col-start-[-1] w-full grid grid-cols-[1fr_2fr] px-1 content-center text-xs sm:text-sm gap-10 font-light sm:font-normal;
}

```

    The`active-paymaster-nav ` class is reusing the same styling with the navbar in the landing page .

To view the paymasters page navigate to `http://localhost:3000/auth/{auth_id}/activepaymaster`

- **allpaymasters** : This is the paymasters list component.
  `activepaymaster/allpaymasters` styling for this component and the media queries is found at `styles/activepaymaster/allpaymasters.module.css`

- **firstSection** : It is the first part i.e first section of the active paymasters page

- **paymastersListPage** : This is where all the components in the active paymaster directory are assembled together and then exported to the active paymasters route.
  `activepaymaster/paymastersListPage` is exported to `/auth/[private]/activepaymaster`

## Auth directory

- **private**: A HOC for authentication using NEXT_PUBLIC_AUTH_ID environmental variable

## landing page

- **firstSection** : This is the hero section of the landing page.

- **landing** : This is where all the components in the landing page directory are assembled together and then exported to the landing page route.
  `activepaymaster/landingpage` is exported to `/auth/[private]`

**Applied padding for windows scaling b/w 100 to 125**

```

/*************scaling 100 to 125***********************/

@media (min-width: 1100px) and (-webkit-min-device-pixel-ratio: 1.25) {
  .scaleto100 {
    @apply px-20;
  }
}


```

`tailwind.config.css` custom configurations

```

 theme: {
    extend: {
      colors: {
        backgroundTheme: "#11142b",
        secondaryGlow: "#7d0cc1",
        jsClientGlow: "#7D0CC1",
        jsClientDark: "rgba(125, 12, 193, 0.3)",
        blurColor: "rgba(125, 12, 193, 0.5)",
        flowerBlur: "rgba(193, 12, 12, 0.5)",
      },
      fontFamily: {
        monserrat: "Montserrat",
        color: "#fff",
      },
      shadow: {
        btn: "4px 4px 14px rgba(0, 0, 0, 0.15)",
      },


```
