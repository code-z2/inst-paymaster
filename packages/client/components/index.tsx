// Reusable exports
export { default as Image } from "next/image";
export { default as Link } from "next/link";

// Landing page components
export { default as LandingPage } from "./landingpage/Default";

export { default as LandingPageLayout } from "./layouts/Default";
export { UnSecureLandingPage } from "./landingpage/Default";
export { default as Hero } from "./landingpage/Hero";
export { default as RebatesMiddleware } from "./landingpage/RebatesMiddleware";
export { default as BuildPaymasters } from "./landingpage/BuildPaymasters";

// authentication HOC component
export { default as Auth } from "./auth/private";

// paymasters list page components
export { default as PaymastersList } from "./activepaymaster/paymastersListPage";
export { default as Loadmore } from "./activepaymaster/loadmore";
export { default as ActiveNav } from "./activepaymaster/activeNav";
export { default as FirstSection } from "./activepaymaster/firstSection";
export { default as Allpaymasters } from "./activepaymaster/allpaymasters";
