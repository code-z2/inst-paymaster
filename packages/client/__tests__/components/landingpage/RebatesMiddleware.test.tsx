import React from "react";
import "@testing-library/jest-dom";
import { RebatesMiddleware } from "../../../components";
import { render } from "@testing-library/react";

/*
 *  snapshots of the middleware section components
 */
describe("Middleware Section ", () => {
  it("Should render the middleware components ", () => {
    const home = render(<RebatesMiddleware />);
    expect(home).toMatchSnapshot();
  });
});
