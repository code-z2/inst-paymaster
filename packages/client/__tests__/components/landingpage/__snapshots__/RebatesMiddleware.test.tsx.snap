// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Middleware Section  Should render the middleware components  1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="rebates-middleware-section site-section"
      >
        <div
          class="wrapper w-image"
        >
          <div
            class="section-content"
          >
            <h2
              class="font-semibold text-5xl"
            >
              Rebates Middleware
            </h2>
            <p>
              Loyalty programs are a way we know to give back to our most loyal users and with account abstraction, it is possible to create a standalone incentive system.
            </p>
            <p>
              Your users can earn a fraction of their transaction value each time they make a transaction.
            </p>
            <p>
              However building this system distinctively may incur additional miscellaneous. With paymasters, you can integrate chargebacks as a middleware.
            </p>
          </div>
          <div
            class="section-img-cont"
          >
            <img
              alt="Rebates Illustration"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
          </div>
        </div>
      </section>
      <section
        class="features-section site-section"
      >
        <div
          class="wrapper"
        >
          <div
            class="features-grid"
          >
            <article
              class="feature-item "
            >
              <img
                alt="Search Icon"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
              <h3
                class="font-semibold text-xl"
              >
                More visible than macro
              </h3>
              <p
                class="text-sm"
              >
                Explore a wide range of Paymasters, find protocols with Paymaster support . Get more value when using dapps.
              </p>
            </article>
            <article
              class="feature-item "
            >
              <img
                alt="Search Icon"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
              <h3
                class="font-semibold text-xl"
              >
                Much more than transactions
                 
              </h3>
              <p
                class="text-sm"
              >
                An aggregation of paymasters for users to have access to Ethereum protocols without incurring gas costs or unnecessary onboarding cost.
              </p>
            </article>
            <article
              class="feature-item "
            >
              <img
                alt="Search Icon"
                data-nimg="1"
                decoding="async"
                height="40"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                style="color: transparent;"
                width="40"
              />
              <h3
                class="font-semibold text-xl"
              >
                UX more superior than superior
              </h3>
              <p
                class="text-sm"
              >
                Create custom paymasters that does what you want or targets a group or community without complexity. Craft innovative transaction flows for your users.
              </p>
            </article>
          </div>
          <div
            class="features-img-cont"
          >
            <img
              alt="Abstract Sphere"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
            <div
              class="circles"
            >
              <img
                alt="Circles"
                data-nimg="1"
                decoding="async"
                height="300"
                loading="lazy"
                src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
                style="color: transparent;"
                width="300"
              />
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="rebates-middleware-section site-section"
    >
      <div
        class="wrapper w-image"
      >
        <div
          class="section-content"
        >
          <h2
            class="font-semibold text-5xl"
          >
            Rebates Middleware
          </h2>
          <p>
            Loyalty programs are a way we know to give back to our most loyal users and with account abstraction, it is possible to create a standalone incentive system.
          </p>
          <p>
            Your users can earn a fraction of their transaction value each time they make a transaction.
          </p>
          <p>
            However building this system distinctively may incur additional miscellaneous. With paymasters, you can integrate chargebacks as a middleware.
          </p>
        </div>
        <div
          class="section-img-cont"
        >
          <img
            alt="Rebates Illustration"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
        </div>
      </div>
    </section>
    <section
      class="features-section site-section"
    >
      <div
        class="wrapper"
      >
        <div
          class="features-grid"
        >
          <article
            class="feature-item "
          >
            <img
              alt="Search Icon"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <h3
              class="font-semibold text-xl"
            >
              More visible than macro
            </h3>
            <p
              class="text-sm"
            >
              Explore a wide range of Paymasters, find protocols with Paymaster support . Get more value when using dapps.
            </p>
          </article>
          <article
            class="feature-item "
          >
            <img
              alt="Search Icon"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <h3
              class="font-semibold text-xl"
            >
              Much more than transactions
               
            </h3>
            <p
              class="text-sm"
            >
              An aggregation of paymasters for users to have access to Ethereum protocols without incurring gas costs or unnecessary onboarding cost.
            </p>
          </article>
          <article
            class="feature-item "
          >
            <img
              alt="Search Icon"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <h3
              class="font-semibold text-xl"
            >
              UX more superior than superior
            </h3>
            <p
              class="text-sm"
            >
              Create custom paymasters that does what you want or targets a group or community without complexity. Craft innovative transaction flows for your users.
            </p>
          </article>
        </div>
        <div
          class="features-img-cont"
        >
          <img
            alt="Abstract Sphere"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
            style="color: transparent;"
            width="300"
          />
          <div
            class="circles"
          >
            <img
              alt="Circles"
              data-nimg="1"
              decoding="async"
              height="300"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=640&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=384&q=75 1x, /_next/image?url=%2Fimg.jpg&w=640&q=75 2x"
              style="color: transparent;"
              width="300"
            />
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
